### PR TITLE
get study plan operation done

### PR DESCRIPTION
**Pull request by Mihai-Cojusnean**
the following operation must be debated

### CODE REVIEW

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.regex.Pattern;
+
+@Component
+public class PlanStudiiParser {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private final String[] PARAMS = {"__EVENTTARGET", "__EVENTARGUMENT", "__LASTFOCUS", "__VIEWSTATE", "__VIEWSTATEGENERATOR", "__EVENTVALIDATION"};
+    private final String PLAN_STUDII_URL = "http://planstudii.usarb.md";
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: What is STUDII meaning?

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import org.springframework.cglib.proxy.UndeclaredThrowableException;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+public class PlanStudiiParser {
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: and here

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    private final OkHttpClient httpClient = new OkHttpClient();
+
+    public void getRequest() {
+        String AGENT_URL = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36";
+        CloseableHttpClient httpClient = HttpClientBuilder.create()
+                .setRedirectStrategy(new LaxRedirectStrategy())
+                .setUserAgent(AGENT_URL)
+                .setKeepAliveStrategy(new DefaultConnectionKeepAliveStrategy())
+                .build();
+        HttpGet httpGet = new HttpGet(PLAN_STUDII_URL);
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: rename to normal name pls

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        try (CloseableHttpResponse response1 = httpClient.execute(httpGet)) {
+            HttpEntity entity1 = response1.getEntity();
+            responseBody = EntityUtils.toString(entity1);
+            EntityUtils.consume(entity1);
+        } catch (UndeclaredThrowableException | IOException ignored) {
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: why catch is empty?

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    }
+
+    public String generateWebRequest(String group) {
+        List<String> dd, ddValue;
+        switch (group.toUpperCase()) {
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: need to find more useful solutions isntead of this switch

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                    .url(PLAN_STUDII_URL)
+                    .addHeader("User-Agent", "OkHttp Bot")
+                    .post(formBody)
+                    .build();
+
+            try (Response response = httpClient.newCall(request).execute()) {
+
+                if (!response.isSuccessful()) throw new IOException("Unexpected code " + response);
+
+                responseBody = response.body().string();
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: Method invocation 'string' may produce 'NullPointerException'

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        return objectMapper.writeValueAsString(sb);
+    }
+
+    private static String getHiddenParam(String id, String body) {
+        Matcher m = Pattern.compile(String.format("id=\"%s\"\\s+value=\"([^\"]+)\"", id)).matcher(body);
+        if (m.find()) {
+            return m.group(1);
+        }
+        return "";
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: use StringUtils.EMPTY instead

File: parsing-service/src/main/java/com/flux/parsingservice/service/ParserService.java [hunk 8]
```diff
@@ -2,16 +2,19 @@
...
 import org.springframework.stereotype.Service;
 
 @Service
 public class ParserService {
 
     private final LessonsParser parser;
+    private final PlanStudiiParser planStudiiParser;
 
-    public ParserService(LessonsParser parser) {
+    public ParserService(LessonsParser parser, PlanStudiiParser planStudiiParser) {
```
Comment (taragrammatonn) [inline parsing-service/src/main/java/com/flux/parsingservice/service/ParserService.java:15]: pls rename this class

File: telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+    public GetPlanStudiiCommandGenerator(RestTemplateService restTemplateService) {
+        this.restTemplateService = restTemplateService;
+    }
+
+    @Override
+    @SneakyThrows
+    public SendMessage generateCommand(Update update) {
+        String userGroup = restTemplateService.getForObject(UserVO.class, GET_USER_BY_CHAT_ID, update.getMessage().getChatId())
+                .getUserGroup();
```
Comment (taragrammatonn) [inline telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java]: use code covections

File: telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        String response = restTemplateService.getForObject(String.class, GET_STUDY_PLAN, userGroup);
+        return new SendMessage(update.getMessage().getChatId(), response);
+    }
+
+    @Override
+    public String getInputCommand() {
+        return "/planStudii";
```
Comment (taragrammatonn) [inline telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java]: add language pack

**Review by taragrammatonn — DISMISSED** [review]

File: telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java (hunk 10 above)
Comment (Mihai-Cojusnean) [reply telegram-user-service/src/main/java/com/flux/telegramservice/service/generator/impl/GetPlanStudiiCommandGenerator.java]: для этого уже есть ветка

File: parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java (hunk 7 above)
Comment (Mihai-Cojusnean) [reply parsing-service/src/main/java/com/flux/parsingservice/parser/PlanStudiiParser.java]: https://stackoverflow.com/questions/4095501/is-stringutils-empty-recommended

**Pull request closed without merging**